staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -132,31 +165,43 @@ type MutableFeatureGate interface {
 type featureGate struct {
 	featureGateName string
 
-	special map[Feature]func(map[Feature]FeatureSpec, map[Feature]bool, bool)
+	special map[Feature]func(map[Feature]VersionedSpecs, map[Feature]bool, bool, *version.Version)
 
 	// lock guards writes to known, enabled, and reads/writes of closed
 	lock sync.Mutex
 	// known holds a map[Feature]FeatureSpec
 	known atomic.Value
 	// enabled holds a map[Feature]bool
 	enabled atomic.Value
+	// enabledRaw holds a map[string]bool of the parsed flag
+	enabledRaw atomic.Value
```

Comment:
`sync.Map` is an option that might be convenient if these are write-once, read-many.

Reply:
These are not exactly write-once. It can be updated whenever `SetFromMap` is called.